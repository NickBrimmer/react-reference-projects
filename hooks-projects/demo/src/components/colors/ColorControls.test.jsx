import React from 'react';
import { shallow } from 'enzyme';
import ColorControls from './ColorControls';

describe('ColorControls component', () => {
  let handleChange; 
  let wrapper;
  beforeEach(() => {
    handleChange = jest.fn();
    wrapper = shallow(<ColorControls
      text="Hello!"
      backgroundColor="lightskyblue"
      textColor="red"
      onChange={handleChange}
    />);
  });

  it('has an input with a value set to the text prop', () => {
    expect(wrapper.find('input[name="text"]').prop('value')).toEqual('Hello!');
  });
  
  it('has an input with a value set to the backgroundColor prop', () => {
    expect(wrapper.find('input[name="backgroundColor"]').prop('value')).toEqual('lightskyblue');
  });
  
  it('has an input with a value set to the textColor prop', () => {
    expect(wrapper.find('input[name="textColor"]').prop('value')).toEqual('red');
  });

  it('invokes an onChange prop when the text input changes', () => {
    // onChange
    wrapper.find('input[name="text"]').simulate('change');
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it('invokes an onChange prop when the backgroundColor input changes', () => {
    wrapper.find('input[name="backgroundColor"]').simulate('change');
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it('invokes an onChange prop when the textColor input changes', () => {
    wrapper.find('input[name="textColor"]').simulate('change');
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it('renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
