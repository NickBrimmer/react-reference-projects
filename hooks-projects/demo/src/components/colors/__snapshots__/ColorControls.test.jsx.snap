// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorControls component renders the component 1`] = `
<Fragment>
  <input
    name="text"
    onChange={[MockFunction]}
    type="text"
    value="Hello!"
  />
  <input
    name="backgroundColor"
    onChange={[MockFunction]}
    type="color"
    value="lightskyblue"
  />
  <input
    name="textColor"
    onChange={[MockFunction]}
    type="color"
    value="red"
  />
</Fragment>
`;
